CHANGELOG.md

```diff
@@ -2,6 +2,58 @@
 
 This document includes a curated changelog for each release. We also publish a changelog as the description of a [GitHub release](https://github.com/nginxinc/nginx-kubernetes-gateway/releases), which, by contrast, is auto-generated and includes links to all PRs that went into the release.
 
+## Release 0.4.0
+
+*July 5, 2023*
+
+This release brings:
+- Extended support of the features of the Gateway API resources.
```

Comment:
"Extended" has special meaning in Gateway API, with core, extended and implementation-specific support for features, so I suggest rewriting it:
```suggestion
- More features of the Gateway API resources.
```